### ISSUE TITLE
Authentication failed.

### ISSUE THREAD

**Issue by pageb018** [GuyLewin/home-assistant-lifetime-fitness#1]
I am trying to authenticate this integration but it's keep failing with an "Invalid authentication" error. 

I have confirmed my user name and password by successfully logging into the Life Time site and the Life time app. 

Is there any special API requirements?

**Comment by GuyLewin** [comment]
The way I implemented the authentication is based on what happens when you login to https://my.lifetime.life/login.html .
If the same credentials work on the above URL but not in the integration - do you mind logging in from the browser and using Dev Tools - Network Capture to find the request to https://api.lifetimefitness.com/auth/v2/login ?
If that exists in your network log - perhaps you can paste the format here (without the username and password) so I can see if it's different than what I used.

**Comment by pageb018** [comment]
So yes, I can login into the my.lifetime.life url with the creds. 

It looks like the payload request is as follows:

{"username":"MYUSERNAME","password":"MYPASS"}

hopefully that helps. 

I think that is what you are after, let me know if you need something else.

**Comment by GuyLewin** [comment]
Thanks for checking.
That's exactly what the integration is sending as well.
Does your password have any special characters? If so - can you check how it's being formatted in the request payload? I see my password is plain so it's not an issue but maybe there's some extra handling to do if you have a ' or a " in your password.

**Comment by pageb018** [comment]
So weird. No special characters and I can see the password in plain text as well. Maybe an ad blocker causing it? 

Strange.

**Comment by GuyLewin** [comment]
I'll try to add a more informative log so we can see the JSON response you're getting from the authentication endpoint. But I don't see a difference between what the browser is doing and what the integration does.

**Comment by pageb018** [comment]
Very strange. Let me know and I will update and test again.

**Comment by GuyLewin** [comment]
@pageb018 Try the new version I just released (1.0.3). After you receive an invalid authentication - go to Configuration -> Settings -> Logs and let me know if you see a relevant message from the Life Time Fitness integration there.

**Comment by pageb018** [comment]
Just gave it a shot. Here is the error in the logs. 

I can see it's telling me a require a password change....but I do not get that message during any other auth. I will update my password in a big and give it a shot. 

```
This error originated from a custom integration.

Logger: custom_components.lifetime_fitness.api
Source: custom_components/lifetime_fitness/api.py:56
Integration: Life Time Fitness (documentation, issues)
First occurred: 12:03:24 PM (1 occurrences)
Last logged: 12:03:24 PM

Received invalid authentication response: {'token': 
```

**Comment by GuyLewin** [comment]
Thanks @pageb018 - but I strongly suggest you delete the `token`, `ssoId` and `partyId` out of the above comment (or all of it in general) since it may be used to authenticate on your behalf (my code uses the `token` to login).
I suggest you change your password and that'll invalidate the above token.

**Comment by GuyLewin** [comment]
Regardless - I see the issue now. I'm comparing the `status` field to '200' but I didn't consider there are other status codes considered as successful. I'll fix that and release a new version. Thanks!

**Comment by pageb018** [comment]
Got it. I removed the info. 

I will test when you update. Really appreciate you debugging this!

**Comment by GuyLewin** [comment]
No problem, thanks for providing all the information!

I released version 1.1.0 in which I rewrote the authentication part. This should handle all different errors / warnings from the API. Closing this ticket since it should be resolved, feel free to reopen if there are further issues!

**Comment by pageb018** [comment]
Working great. Thanks again!
